src/StateSmith.Cli/Run/RunHandler.cs

```diff
@@ -98,6 +99,37 @@ private void RunInner()
 
         ReadPastRunInfoDatabase();
         var scanResults = Finder.Scan(searchDirectory: searchDirectory);
+
+        RunInnerInner(searchDirectory, scanResults);
+
+        if( _runHandlerOptions.Watch) {
+            var watchers = new List<FileSystemWatcher>();
+            foreach (var diagramFile in scanResults.targetDiagramFiles.Union( scanResults.targetCsxFiles ))
+            {
+                var watcher = new FileSystemWatcher();
+                watchers.Add(watcher); // don't let watchers go out of scope yet
+                var path = Path.GetDirectoryName(diagramFile);
+                watcher.Path = path!=null && path.Length>0 ? path : "."; 
+                watcher.Filter = Path.GetFileName(diagramFile);
+                _runConsole.WriteLine($"Watching {diagramFile}");
+                _runConsole.WriteLine($"Path: {watcher.Path}");
+                _runConsole.WriteLine($"Filter: {watcher.Filter}");
+                watcher.Changed += (sender, e) => 
+                {
+                    _runConsole.WriteLine($"File {diagramFile} has changed.");
+                    // TODO only process the changed file. Not sure how to break up ScanResults to do this
```

Comment:
if we just poll, or inspect the file extension, we could call the relevant handler [for .csx](https://github.com/StateSmith/StateSmith/blob/7206f7e387f09eefcaf4847609bc55f34d261a28/src/StateSmith.Cli/Run/RunHandler.cs#L232) and we could refactor this [bit of code for diagrams](https://github.com/StateSmith/StateSmith/blob/7206f7e387f09eefcaf4847609bc55f34d261a28/src/StateSmith.Cli/Run/DiagramRunner.cs#L86-L111) into a public method.

Note that the scan will pick up diagrams that are processed by .csx files. Here's where this check is currently done: https://github.com/StateSmith/StateSmith/blob/7206f7e387f09eefcaf4847609bc55f34d261a28/src/StateSmith.Cli/Run/DiagramRunner.cs#L49-L59